osu.Game.Rulesets.Mania/Skinning/ManiaLegacySkinTransformer.cs

```diff
@@ -95,30 +126,13 @@ public Drawable GetDrawableComponent(ISkinComponent component)
             return null;
         }
 
-        private Drawable getResult(GameplaySkinComponent<HitResult> resultComponent)
+        private Drawable getResult(HitResult result)
         {
-            switch (resultComponent.Component)
-            {
-                case HitResult.Miss:
-                    return this.GetAnimation("mania-hit0", true, true);
-
-                case HitResult.Meh:
-                    return this.GetAnimation("mania-hit50", true, true);
-
-                case HitResult.Ok:
-                    return this.GetAnimation("mania-hit100", true, true);
-
-                case HitResult.Good:
-                    return this.GetAnimation("mania-hit200", true, true);
+            string image = GetConfig<ManiaSkinConfigurationLookup, string>(
```

Comment:
i don't know how to read this line. maybe needs some commenting and splitting out.

also "image" is a bad variable name for this, it's more likely a "filename"?

Reply:
https://github.com/ppy/osu/blob/ca2d0a6ea2de85313c20f5dc59dfcccd7c1aa15f/osu.Game.Rulesets.Mania/Skinning/LegacyManiaElement.cs#L21

Looking back at this piece of code, extension methods seems a lot better than what we have now -- We can then refactor that to use it. That would need another PR imo because it would touch some other unrelated files.

Reply:
Change variable `image` to `filename` anyway